pytorch/training/docker/2.0/py3/cu118/Dockerfile.gpu

```diff
@@ -389,22 +372,13 @@ ARG PT_TORCHVISION_URL
 ARG PT_TORCHAUDIO_URL
 ARG PT_TORCHDATA_URL
 
-# Install AWS-PyTorch and other torch packages
-RUN pip uninstall -y torch torchvision torchaudio torchdata \
- && pip install --no-cache-dir -U ${PT_SM_TRAINING_URL} ${PT_TORCHVISION_URL} ${PT_TORCHAUDIO_URL} ${PT_TORCHDATA_URL} torchnet
-
-# Install Nvidia Apex (needs pytorch)
-## Pin apex commit requested by sm-model-parallel team
-RUN git clone https://github.com/NVIDIA/apex && \
-    cd apex && \
-    git checkout aa756ce && \
-    pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
-
-# Install Horovod (needs pytorch)
-RUN pip uninstall -y horovod \
- && ldconfig /usr/local/cuda-11.7/targets/x86_64-linux/lib/stubs \
- && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_CUDA_HOME=/usr/local/cuda-11.7 HOROVOD_WITH_PYTORCH=1 pip install --no-cache-dir horovod==${HOROVOD_VERSION} \
- && ldconfig
+# force remove to prevent removing all packages depending on pytorch
+RUN /opt/conda/bin/conda remove --force pytorch \
```

Comment:
again, what's the purpose of force remove pytorch and install it again?

Reply:
This is for removing the ec2 package and installing the sm channel package. --force flags makes sure no packages that depend on pytorch are removed